### PR TITLE
Fix failure for pv_eoi hyper_features since RHEL-8.1

### PR DESCRIPTION
**Pull request by yanjing1993**
The failure is caused by the qemu cmd line for pv_eoi hyper_features
with disable/enable has changed from -/+ to off/on

Signed-off-by: JingYan <jiyan@redhat.com>

### CODE REVIEW

**Comment by yanjing1993** [comment]
Tested this issue on both RHEL-7 and RHEL-8; all passed.

# avocado run --vt-type libvirt --vt-machine-type q35 libvirt_qemu_cmdline.hypervisor_features.pv_eoi.enable --vt-connect-uri qemu:///system
JOB ID     : 46ff9ec24d11a5f82cb32593ecead68eab42708f
JOB LOG    : /root/avocado/job-results/job-2019-08-28T04.32-46ff9ec/job.log
 (1/1) type_specific.io-github-autotest-libvirt.libvirt_qemu_cmdline.hypervisor_features.pv_eoi.enable: PASS (7.35 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 9.43 s

# avocado run --vt-type libvirt --vt-machine-type q35 libvirt_qemu_cmdline.hypervisor_features.pv_eoi.disable --vt-connect-uri qemu:///system
JOB ID     : eb31700184921de58ddc0ab129b45ae78f8cdc9c
JOB LOG    : /root/avocado/job-results/job-2019-08-28T04.32-eb31700/job.log
 (1/1) type_specific.io-github-autotest-libvirt.libvirt_qemu_cmdline.hypervisor_features.pv_eoi.disable: PASS (7.53 s)
RESULTS    : PASS 1 | E

File: libvirt/tests/src/libvirt_qemu_cmdline.py [hunk 1]
```diff
@@ -25,13 +28,24 @@ def config_feature_pv_eoi(test, vmxml, **kwargs):
...
     eoi_enable = kwargs.get('eoi_enable', 'on')
-    if eoi_enable == 'on':
-        qemu_flags.append('+kvm_pv_eoi')
-    elif eoi_enable == 'off':
-        qemu_flags.append('-kvm_pv_eoi')
-    else:
-        logging.error("Invaild value %s, eoi_enable must be 'on' or 'off'",
-                      eoi_enable)
+    get_hostos_version = astring.to_text(process.run("cat /etc/redhat-release", shell=True).stdout)
+    if re.search(r'(\d+(\.\d+)?)', get_hostos_version) is not None:
```
Comment (chunfuwen) [inline libvirt/tests/src/libvirt_qemu_cmdline.py]: what happen if re.search return none ? that means it will skip all qemu_flags options initialization

Comment (yanjing1993) [reply libvirt/tests/src/libvirt_qemu_cmdline.py]: updated and retesting also passed.

File: libvirt/tests/src/libvirt_qemu_cmdline.py [hunk 2]
```diff
@@ -25,13 +28,26 @@ def config_feature_pv_eoi(test, vmxml, **kwargs):
...
     qemu_flags = []
     eoi_enable = kwargs.get('eoi_enable', 'on')
-    if eoi_enable == 'on':
-        qemu_flags.append('+kvm_pv_eoi')
-    elif eoi_enable == 'off':
-        qemu_flags.append('-kvm_pv_eoi')
-    else:
-        logging.error("Invaild value %s, eoi_enable must be 'on' or 'off'",
-                      eoi_enable)
+    get_hostos_version = astring.to_text(process.run("cat /etc/redhat-release", shell=True).stdout)
```
Comment (Yingshun) [inline libvirt/tests/src/libvirt_qemu_cmdline.py]: I'm not sure if we need to cover other distribution in auto script. If yes, I think distro.detect() whould be better choice to get name, version and release.

File: libvirt/tests/src/libvirt_qemu_cmdline.py [hunk 3]
```diff
@@ -25,13 +28,26 @@ def config_feature_pv_eoi(test, vmxml, **kwargs):
...
+                logging.error("Invaild value %s, eoi_enable must be 'on' or 'off'", eoi_enable)
+        elif hostos_version > float(8.0):
+            if eoi_enable == 'on':
+                qemu_flags.append('kvm-pv-eoi=on')
+            elif eoi_enable == 'off':
+                qemu_flags.append('kvm-pv-eoi=off')
+            else:
+                logging.error("Invaild value %s, eoi_enable must be 'on' or 'off'", eoi_enable)
+        else:
+            test.fail("Can not decide the expected qemu cmd line because of no expected hostos version")
```
Comment (Yingshun) [inline libvirt/tests/src/libvirt_qemu_cmdline.py]: It should raise test.error or test.cancel.

File: libvirt/tests/src/libvirt_qemu_cmdline.py (hunk 2 above)
Comment (yanjing1993) [reply libvirt/tests/src/libvirt_qemu_cmdline.py]: > I'm not sure if we need to cover other distribution in auto script. If yes, I think distro.detect() whould be better choice to get name, version and release.

Hi there are some changes which caused the differences between RHEL-7 and RHEL-8, so in this case, I covered different distribution. I tried the method you mentioned, the following is I get, it seems can not give the release and version:

2019-10-10 06:13:38,139 libvirt_qemu_cmd L0034 DEBUG| <LinuxDistro: name=rhel, version=0, release=0, arch=x86_64>
2019-10-10 06:13:38,140 libvirt_qemu_cmd L0035 DEBUG| <class 'avocado.utils.distro.LinuxDistro'>

**Pull request merged**